app/models/user.rb

```diff
@@ -63,6 +63,7 @@ class User < ActiveRecord::Base
   }
   scope :visible,         -> { except_hidden }
   scope :completer_scope, ->(opts) { visible }
+  scope :puppet_error_notifications, -> { User.includes(:user_mail_notifications).where(:user_mail_notifications => {:interval => 'Subscribe to all hosts'} ) }
```

Comment:
I realize there's a lot of mailing code (legacy I'd say) in the user model, however I don't think mail notifications are part of the 'User' model. If you subclass as I mention before, you can use `MailNotification::PuppetError.all` instead of this scope, `puppet_error_notifications` behavior belong in there better.